packages/strapi-plugin-i18n/admin/src/utils/getDefaultLocale.js

```diff
@@ -0,0 +1,58 @@
+const hasLocalePermission = (permission, localeCode) => {
+  if (permission) {
+    const hasPermission = permission.properties.locales.includes(localeCode);
+
+    if (hasPermission) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+const getFirstLocale = permission => {
+  if (permission) {
+    const firstAuthorizedForReadNonDefaultLocale = permission.properties.locales[0];
+
+    if (firstAuthorizedForReadNonDefaultLocale) {
+      return firstAuthorizedForReadNonDefaultLocale;
+    }
+  }
+
+  return null;
+};
+
+/**
+ * Entry point of the module
+ */
+const getDefaultLocale = (contentType, userPermissions, locales = []) => {
```

Comment:
ℹ️ this is the important part

Reply:
Here you don't need the content type as you will received all the permissions related to the content type.

`const getDefaultLocale = (contentTypeRelatedPermissions, locales)`